pkg/systemd/overview-cards/shutdownStatus.jsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState, useEffect } from 'react';
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
+import { PowerOffIcon, RedoIcon } from "@patternfly/react-icons";
+
+import * as timeformat from "timeformat";
+
+import cockpit from "cockpit";
+
+const _ = cockpit.gettext;
+const systemd_file = "/run/systemd/shutdown/scheduled";
+
+const getScheduledShutdown = (setShutdownTime, setShutdownType) => {
+    const client = cockpit.dbus("org.freedesktop.login1");
+    return client.call("/org/freedesktop/login1", "org.freedesktop.DBus.Properties", "Get", ["org.freedesktop.login1.Manager", "ScheduledShutdown"], { type: "ss" }).then(reply => {
+        const result = reply[0];
+        if (result) {
+            setShutdownType(result.v[0]);
+            setShutdownTime(result.v[1]);
+        }
+    });
+};
+
+const cancelShutdownAction = () => {
+    const client = cockpit.dbus("org.freedesktop.login1");
+    return client.call("/org/freedesktop/login1", "org.freedesktop.login1.Manager", "CancelScheduledShutdown").then(reply => {
+        if (!reply[0]) {
+            console.error("Unable to cancel shutdown");
+        }
+    });
+};
+
+export const ShutDownStatus = () => {
+    const [shutdownType, setShutdownType] = useState(null);
+    const [shutdownTime, setShutdownTime] = useState(0);
+
+    useEffect(() => {
+        if (shutdownType === null) {
+            getScheduledShutdown(setShutdownTime, setShutdownType);
+
+            // logind does not have a propertieschanged mechanism https://github.com/systemd/systemd/issues/22244
+            cockpit.file(systemd_file).watch(content => {
+                getScheduledShutdown(setShutdownTime, setShutdownType);
+            });
+        }
+    }, [shutdownType]);
+
+    // We only care about these two types
+    if (shutdownType !== "poweroff" && shutdownType !== "reboot") {
+        return null;
```

Comment:
We don't really expect any others, do we? Please make this a bit stronger and console.warn() about unknown ones, so that this is easier to debug. Also "care about" → "know of"?

Reply:
There are others, such as "halt" which we don't support but maybe should show?

Reply:
Showing the additional ones can be a follow-up for sure. So maybe console.log() for now?